File: app/src/main/java/com/recursia/popularmovies/data/mappers/EntityToTrailerDatabaseModelMapper.java
package com.recursia.popularmovies.data.mappers;

import com.recursia.popularmovies.data.models.TrailerDatabaseModel;
import com.recursia.popularmovies.domain.models.Trailer;
import com.recursia.popularmovies.utils.Mapper;

public class EntityToTrailerDatabaseModelMapper extends Mapper<Trailer, TrailerDatabaseModel> {
    @Override
    public TrailerDatabaseModel transform(Trailer trailer) {
        TrailerDatabaseModel trailerDatabaseModel = new TrailerDatabaseModel();
        trailerDatabaseModel.setId(trailer.getId());
        trailerDatabaseModel.setKey(trailer.getKey());
        trailerDatabaseModel.setMovieId(trailer.getMovieId());
        trailerDatabaseModel.setName(trailer.getName());
        trailerDatabaseModel.setSite(trailer.getSite());
        trailerDatabaseModel.setSize(trailer.getSize());
        trailerDatabaseModel.setType(trailer.getType());
        return trailerDatabaseModel;
    }
}

File: app/src/main/java/com/recursia/popularmovies/data/mappers/EntityToTrailerDatabaseModelMapper.kt
package com.recursia.popularmovies.data.mappers

import com.recursia.popularmovies.data.models.TrailerDatabaseModel
import com.recursia.popularmovies.domain.models.Trailer
import com.recursia.popularmovies.utils.Mapper

class EntityToTrailerDatabaseModelMapper : Mapper<Trailer, TrailerDatabaseModel>() {
    override fun transform(trailer: Trailer): TrailerDatabaseModel {
        val trailerDatabaseModel = TrailerDatabaseModel()
        trailerDatabaseModel.id = trailer.id
        trailerDatabaseModel.key = trailer.key
        trailerDatabaseModel.movieId = trailer.movieId
        trailerDatabaseModel.name = trailer.name
        trailerDatabaseModel.site = trailer.site
        trailerDatabaseModel.size = trailer.size
        trailerDatabaseModel.type = trailer.type
        return trailerDatabaseModel
    }
}
